subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/tasks/internal/ValidateAction.java

```diff
@@ -131,7 +131,7 @@ private static DefaultTypeValidationContext createValidationContext(Class<?> top
     }
 
     private static void validateCacheabilityAnnotationPresent(Class<?> topLevelBean, boolean cacheable, Class<? extends Annotation> cacheableAnnotationClass, DefaultTypeValidationContext validationContext) {
-        if (!cacheable && topLevelBean.getAnnotation(DisableCachingByDefault.class) == null) {
+        if (!topLevelBean.isInterface() && !cacheable && topLevelBean.getAnnotation(DisableCachingByDefault.class) == null) {
```

Comment:
I think it would make sense to do this in a separate `if`, something like:

```java
if (topLevelBean.isInterface()) {
    // Won't validate interfaces
    return;
}
```

Reply:
Done.